### ISSUE TITLE
Element not scrollable

### ISSUE THREAD

**Issue by doque** [cypress-io/cypress#5022]
### Current behavior:

We're trying to scroll down a page in order to trigger lazy-loading of content. This does not seem to work. While debugging, I found that scrolling on an arbitrary web page is also not possible:

`cy.scrollTo("bottom")` throws an error:

```
CypressError: Timed out retrying: cy.scrollTo() failed because this element is not scrollable:
```

### Desired behavior:

The selected element is scrolled

### Steps to reproduce: (app code and test code)
Tried this on 3.4.1 and using `https://wikipedia.org` as `baseUrl` in `cypress.json`

```
it('scroll', () => {
    cy.visit('https://www.wikipedia.org');
    cy.scrollTo('bottom');
});
```

### Versions
Mac OS 10.14.6 (18G87)
Cypress 3.4.1
Electron

**Comment by Mugane** [comment]
I'm getting this same issue when there is no scrollbar for the window. The screen capture should just not scroll in this case, instead of asserting that it cannot be done.

**Comment by jennifer-shehane** [comment]
@Mugane Can you provide a reproducible example of this when taking screenshots?

**Comment by Mugane** [comment]
```
describe('Demo scrollTo fail', () => {
	it('Google screenshot', () => {
		cy.visit('https://www.google.com').then(() => {
			cy.window().scrollTo('bottom')
			.screenshot()
		})
	})
})
```

**Comment by Mugane** [comment]
![image](https://user-images.githubusercontent.com/13005927/116896059-d8f7df00-ac01-11eb-91c7-8460f949fae1.png)

**Comment by jennifer-shehane** [comment]
@Mugane The element in this case is not scrollable since the height of the page does not fill the entire viewport. If you make the viewport a smaller size that requires scrolling - making the window scrollable, then it will scroll. 

```js
describe('Demo scrollTo fail', () => {
  it('Google screenshot', () => {
    cy.viewport(200, 200)
    cy.visit('https://www.google.com')
    cy.window().scrollTo('bottom')
      .screenshot()
  })
})
```
<img width="557" alt="Screen Shot 2021-05-03 at 12 30 41 PM" src="https://user-images.githubusercontent.com/1271364/116910558-6986ed00-ac0b-11eb-8e67-91ba5f8f9b84.png">

**Comment by Mugane** [comment]
Right, that strikes me as a false positive. A fail should not be asserted, the scrollTo should just not do anything. It is completely reasonable to have situations where it is impossible to know if the content will be sufficient to have a scroll bar, and a general solution is required (making the viewport smaller as suggested here will only work for google which is just an example)

**Comment by Mugane** [comment]
Or alternatively, if a parameter can be passed to allow for content that is not scrollable that would also work...

**Comment by jennifer-shehane** [comment]
@Mugane There's an `ensureScrollable ` option you can pass to `scrollTo` so that Cypress ignores the scrollability of the element. This is mentioned in the error message for ignoring this check.

```js
describe('Demo scrollTo fail', () => {
	it('Google screenshot', () => {
		cy.visit('https://www.google.com')
		cy.window().scrollTo('bottom', { ensureScrollable: false })
		.screenshot()
	})
})
```

**Comment by Mugane** [comment]
Thanks!

**Comment by konstantinschuette** [comment]
![image](https://user-images.githubusercontent.com/18708105/155984546-dd388409-ff85-4dbb-96f3-c082d5019ca9.png)
not working for me.

I modified it slightly:
```
it('Google screenshot', () => {
    cy.viewport(200, 200);
    cy.visit('https://www.google.com');
    cy.scrollTo('bottom');
    cy.wait(1000);
  });
```

I'm currently running `Cypress 8.7.0`

**Comment by h1v3m1nd0brutus** [comment]
same here. <window> not scrollable

**Comment by andrii-lundiak** [comment]
Long time I couldn't realize why my tests fails, and for some reason my application use top area of page (`d3` elements) and elements from bottom area (Angular components + D3) is either not rendered or Cypress can't reach it (main reason of test FAIL).

Anyway...
Jan-2023, maybe someone will be interesting that I can confirm workaround using `.viewport()` and `.screenshot()`, which is oddly but helps me.

My initial screen size `1280, 1024` and looks like even change from same size to same size does a trick :) 

```
cy.visit(myRoute);
// my custom wait code

// BELOW CODE DID NOT scroll whatsoever
//
// cy.get('mat-table').eq(0).scrollTo('center', { ensureScrollable: false });
// cy.scrollTo('bottom', { duration: 10000, easing: 'swing' });
// cy.scrollTo('center')
// cy.get('mat-table').scrollIntoView() 
// cy.window().scrollTo('center')
// cy.window().scrollTo('center').screenshot();
// cy.window().scrollTo('bottom', { ensureScrollable: false });
//
// ABOVE DID NOT scroll whatsoever
```

Then I did this way:

```
cy.visit(myRoute);
// my custom wait code

cy.viewport(1280, 1024); // HELPful
cy.wait(2000); // HELPful. But if 500 then it doesn't help
cy.window().scrollTo('center').screenshot(); // Kinda helpful
cy.viewport(1280, 1024); // HELPful
```

It looks like helpful is EITHER using small wait time `600-1500` and `cy.screenshot()` or no `cy.screenshot()` usage but wait time between `3000-5000`. And `.screenshot()` seems to be making 3 screens (doesn't matter of previous wait time `1500, 2500 or 3000`, weirdo...) And I still don't get why Cypress can't scroll properly...

And then my test continue the flow and PASSED.

**Comment by andrii-lundiak** [comment]
As alternate approach (but it didn't help actually) I tried to `cy.get('.my-element-which-is-not-rendered-long-time')` so that I could rely on Cypress [retry-ability](https://docs.cypress.io/guides/core-concepts/retry-ability). But even if all elements found in DOM, that one which I'm targeting is "hidden", at least Cypress can't find it. So scroll behavior is needed for me anyway.
